featuregate/flag.go

```diff
@@ -10,11 +10,31 @@ import (
 	"go.uber.org/multierr"
 )
 
+const (
+	featureGatesFlag            = "feature-gates"
+	featureGatesFlagDescription = "Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature."
+)
+
 // NewFlag returns a flag.Value that directly applies feature gate statuses to a Registry.
+// Deprecated: Use RegisterFlags instead.
 func NewFlag(reg *Registry) flag.Value {
+	return newFeatureGateValue(reg)
+}
+
+func newFeatureGateValue(reg *Registry) flag.Value {
 	return &flagValue{reg: reg}
 }
 
+// FlagsOption is an option for RegisterFlags.
+type FlagsOption interface {
+	private()
+}
+
+// RegisterFlags that directly applies feature gate statuses to a Registry.
+func RegisterFlags(flagSet *flag.FlagSet, reg *Registry, opts ...FlagsOption) {
+	flagSet.Var(newFeatureGateValue(reg), featureGatesFlag, featureGatesFlagDescription)
+}
```

Comment:
@mx-psi sorry for this :).

Do you think we can simplify this API and make this a func on the Registry struct? Makes things simpler. 

Also separate concern: FlagsOption -> RegisterFlagsOption

Reply:
Tried that out on f20fbf9e80216aa468b541614b1cdc494856b36e, looks nice :)